pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,446 @@
+	if err != nil {
+		return nil, err
+	}
+	// if there is a logVerbosity field, input the JSON data to the verbosityMap
+	if err := json.Unmarshal(lvJSON, &verbosityMap); err != nil {
+		return nil, err
+	}
+	return verbosityMap, nil
+}
+
+func createOutputLines(verbosityVal map[string]uint, options map[string]uint) []string {
+	var lines []string
+	allIsSet := false
+	if _, exist := options["all"]; exist {
+		allIsSet = true
+	}
+	for component := VirtAPI; component < All; component++ { // all is the last component, and do not need to check it
+		componentName := GetComponentNameByVirtComponent(component)
+		JSONName := getJSONNameByVirtComponent(component)
+		if _, exist := options[componentName]; exist || allIsSet {
+			line := fmt.Sprintf("%s=%d", componentName, verbosityVal[JSONName])
+			lines = append(lines, line)
+		}
+	}
+	return lines
+}
+
+func createShowMessage(currentLv, options map[string]uint) []string {
+	// fill the unattended verbosity with default verbosity
+	// key: JSONName, value: verbosity
+	var verbosityVal = map[string]uint{
+		"virtAPI":        virtconfig.DefaultVirtAPILogVerbosity,
+		"virtController": virtconfig.DefaultVirtControllerLogVerbosity,
+		"virtHandler":    virtconfig.DefaultVirtHandlerLogVerbosity,
+		"virtLauncher":   virtconfig.DefaultVirtLauncherLogVerbosity,
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// update the verbosity based on the existing verbosity in the KubeVirt CR
+	for key, value := range currentLv {
+		verbosityVal[key] = value
+	}
+
+	// create a message to show verbosity for the specified component
+	lines := createOutputLines(verbosityVal, options)
+
+	return lines
+}
+
+func addPatch(patchData *[]patch.PatchOperation, currentLv map[string]uint) {
+	*patchData = append(*patchData, patch.PatchOperation{
+		Op:    patchOperation,
+		Path:  patchPath,
+		Value: currentLv,
+	})
+}
+
+func setVerbosity(currentLv, options map[string]uint, patchData *[]patch.PatchOperation) {
+	// update currentLv based on the user-specified verbosity for all components
+	if verbosity, exist := options["all"]; exist {
+		for component := VirtAPI; component < All; component++ {
+			JSONName := getJSONNameByVirtComponent(component)
+			currentLv[JSONName] = verbosity
+		}
+	}
+
+	// update currentLv based on the user-specified verbosity for each component
+	for componentName, verbosity := range options {
+		if componentName == "all" {
+			continue
+		}
+		JSONName := getJSONNameByComponentName(componentName)
+		currentLv[JSONName] = verbosity
+	}
+
+	// in case of just reset (no set operation after the reset), don't need to add another patch
+	if len(currentLv) != 0 {
+		addPatch(patchData, currentLv)
+	}
+}
+
+func createPatch(currentLv, options map[string]uint) ([]byte, error) {
+	patchData := []patch.PatchOperation{}
+
+	if isReset {
+		// reset only if verbosity exists, otherwise do nothing
+		if len(currentLv) != 0 {
```

Comment:
You can reduce one indentation by doing a single if with `if isReset && len(currentLv) != 0 {`

Reply:
Great. Thanks.